onebusaway-android/src/main/java/org/onebusaway/android/ui/HomeActivity.java

```diff
@@ -2001,4 +2008,22 @@ private boolean isSlidingPanelCollapsed() {
     public ArrivalsListFragment getArrivalsListFragment() {
         return mArrivalsListFragment;
     }
+
+    private void checkBatteryOptimizations() {
+        if (Application.isIgnoringBatteryOptimizations(getApplicationContext())) {
+            showIgnoreBatteryOptimizationDialog();
+        }
+    }
+
+    private void showIgnoreBatteryOptimizationDialog() {
+        new android.app.AlertDialog.Builder(this)
+                .setMessage(R.string.application_ignoring_battery_opt_message)
+                .setTitle(R.string.application_ignoring_battery_opt_title)
+                .setIcon(R.drawable.ic_alert_warning)
+                .setPositiveButton(R.string.ok,
```

Comment:
This should be a yes/no dialog asking the user for permission to allow OneBusAway to run in the background and collect travel behavior data (if they get to this point they've opt-ed into the study but blacklisted OBA for running in the background, intentionally or unintentionally).  If yes, we should fire the `ACTION_REQUEST_IGNORE_BATTERY_OPTIMIZATIONS ` Intent (requires `REQUEST_IGNORE_BATTERY_OPTIMIZATIONS` permission) to trigger a system dialog to let the user whitelist OBA directly without going to the settings.

If no, we shouldn't prompt them about this again.

See https://developer.android.com/training/monitoring-device-state/doze-standby#support_for_other_use_cases for more details on the Intent to ask the user to whitelist the app.

Reply:
@barbeau Implemented this feature in this https://github.com/OneBusAway/onebusaway-android/pull/988/commits/78d3bec3a078008a60042ea28b021e0b607a2a7c commit.  Screenshots look like this:

![device-2019-07-02-160622](https://user-images.githubusercontent.com/2777974/60552630-64da1080-9ce4-11e9-8136-91ac85b969b1.png) | ![device-2019-07-02-160647](https://user-images.githubusercontent.com/2777974/60552629-64da1080-9ce4-11e9-9dac-dd7fd13b8ccc.png)

Reply:
After further discussion, based on testing to date we haven't needed to whitelist the app yet and we seem to be getting reasonable data, so we may not want to do this for travel behavior data collection, as it could have negative consequences on battery life, causing people to opt-out.  We're going to implement this in this PR but I'll remove in a subsequent commit in the master branch so we can re-add the feature quickly if needed based on further beta testing.